habitat/sims/habitat_simulator/habitat_simulator.py

```diff
@@ -270,6 +270,11 @@ def __init__(self, config: Config) -> None:
         self.sim_config = self.create_sim_config(self._sensor_suite)
         self._current_scene = self.sim_config.sim_cfg.scene_id
         super().__init__(self.sim_config)
+        # load additional object paths specified by the dataset
+        # TODO: Should this be moved elsewhere?
+        obj_attr_mgr = self.get_object_template_manager()
+        for path in self.habitat_config.ADDITIONAL_OBJECT_PATHS:
```

Comment:
The need for `additional_object_paths` feels a bit hacky. Why is this needed in practice? Why can't the ReplicaCAD scene dataset include all the objects we need?

Are we going to need `additional_urdf_paths`, `additional_fairmotion_paths`, etc.  in the future?

Reply:
A single SceneDataset is self-contained. In this case we are loading scenes from ReplicaCAD, but we are adding clutter objects from YCB. Since these are separate SceneDatasets and we don't want to contrive some explicit merging, we need to load multiple data sources into the same `MetaDataMediator` instance. Adding this field to the generator config is the most lightweight option. 

Your extrapolation to adding multiple fields for merging different types of data into the active SceneDataset programmatically makes sense and though it may not be pretty, it would probably still be the most lightweight option if those additional types were needed. Another option would be to simply load multiple SceneDatasets into the same MetaDataMediator. In this case we would need to swap the active SceneDataset depending on the source of each asset we attempt to load (which would be cumbersome), or merge multiple SceneDatasets into a single programmatic data-structure. Perhaps @jturner65 can weigh in, but I think the latter could be challenging due to potential issues with naming collisions.

Reply:
One possible solution would be to have an extra scene dataset configuration defining the paths to all these "extra" asset paths.  The MM is designed to support loading the metadata for multiple datasets and having them all available at one time.  The user could then just switch datasets on demand to acquire the configurations desired to instantiate the various objects they require.

Reply:
I agree that from a user perspective the cleanest option at this time is to manually create your own SceneDataset merging all the assets you want to have available.

I also agree that the most likely version of this paradigm to support the current application would be an extra SceneDataset config file which externally references assets and metadata from other datasources as @jturner65 suggests. This would allow us to avoid re-packaging and re-distributing composite datasets which could have conflicting license/use-age terms. For example, we could bundle `data_source` dependencies with the top level metadata file in a `data_group` within the `datasets_download.py` utility.